internal/engine/worker/atproto/core/worker.go

```diff
@@ -0,0 +1,285 @@
+		zap.L().Warn("unsupported type", zap.String("type", atprotoTask.Message.Collection))
+
+		return nil, nil
+	}
+
+	if len(activity.Actions) == 0 {
+		return nil, nil
+	}
+
+	return activity, nil
+}
+
+// transformPost transforms a post message into an activity.
+func (w *worker) transformPost(_ context.Context, message at.Message, activity *activityx.Activity) {
+	activity.Type = typex.SocialPost
+	activity.From = message.Did.String()
+	activity.To = message.Did.String()
+
+	activity.Actions = []*activityx.Action{
+		w.buildPostAction(activity.From, activity.To, activity.Type, w.buildPostMetadata(message)),
+	}
+}
+
+// transformComment transforms a comment message into an activity.
+func (w *worker) transformComment(_ context.Context, message at.Message, activity *activityx.Activity) {
+	activity.Type = typex.SocialComment
+	activity.From = message.Did.String()
+	activity.To = message.RefMessage.Did.String()
+
+	post := w.buildPostMetadata(message)
+
+	post.Target = w.buildPostMetadata(lo.FromPtr(message.RefMessage))
+
+	activity.Actions = []*activityx.Action{
+		w.buildPostAction(activity.From, activity.To, activity.Type, post),
+	}
+}
+
+// transformRepost transforms a repost message into an activity.
+func (w *worker) transformRepost(_ context.Context, message at.Message, activity *activityx.Activity) {
+	activity.Type = typex.SocialShare
+	activity.From = message.Did.String()
+	activity.To = message.RefMessage.Did.String()
+
+	activity.Actions = []*activityx.Action{
+		w.buildPostAction(activity.From, activity.To, activity.Type, w.buildPostMetadata(lo.FromPtr(message.RefMessage))),
+	}
+}
+
+// transformLike transforms a like message into an activity.
+func (w *worker) transformLike(_ context.Context, message at.Message, activity *activityx.Activity) {
+	activity.Type = typex.SocialLike
+	activity.From = message.Did.String()
+	activity.To = message.RefMessage.Did.String()
+
+	activity.Actions = []*activityx.Action{
+		w.buildPostAction(activity.From, activity.To, activity.Type, w.buildPostMetadata(lo.FromPtr(message.RefMessage))),
+	}
+}
+
+// transformProfile transforms a profile message into an activity.
+func (w *worker) transformProfile(_ context.Context, message at.Message, activity *activityx.Activity) {
+	activity.Type = typex.SocialProfile
+	activity.From = message.Did.String()
+	activity.To = message.Did.String()
+
+	activity.Actions = []*activityx.Action{
+		{
+			Tag:      activity.Type.Tag(),
+			Type:     activity.Type,
+			Platform: w.Platform(),
+			From:     activity.From,
+			To:       activity.To,
+			Metadata: w.buildProfileMetadata(message),
+		},
+	}
+}
+
+// buildPostMetadata constructs metadata for a post message.
+func (w *worker) buildPostMetadata(message at.Message) *metadata.SocialPost {
+	post := &metadata.SocialPost{
+		Handle:        message.Handle,
+		ProfileID:     message.Did.String(),
+		PublicationID: message.Rkey,
+		ContentURI:    message.URI,
+		Timestamp:     uint64(message.CreatedAt.Unix()),
```

Comment:
When message.CreatedAt is set to 0001-01-01 00:00:00 +0000, it should default to using message.Feed.CreatedAt

Reply:
`message.CreatedAt` is the result of parsing `message.Feed.CreatedAt`